lib/kino_db/connection_cell.ex

```diff
@@ -277,4 +279,25 @@ defmodule KinoDB.ConnectionCell do
       nodes -> {:__block__, [], nodes}
     end
   end
+
+  defp help_box(%{"type" => "bigquery"}) do
+    if Code.ensure_loaded?(Mint.HTTP) do
+      if is_running_on_google_metadata?() do
+        "You are running inside Google Cloud. Uploading the credentials above is optional."
+      else
+        ~s|You must upload your Google BigQuery Credentials (<a href="https://cloud.google.com/iam/docs/creating-managing-service-account-keys" target="_blank">find them here</a>) or authenticate your machine with <strong>gcloud</strong> CLI authentication.|
+      end
+    end
+  end
+
+  defp help_box(_ctx), do: nil
+
+  defp is_running_on_google_metadata? do
+    with {:ok, conn} <- Mint.HTTP.connect(:http, "metadata.google.internal", 80),
+         {:ok, _} <- Mint.HTTP.set_mode(conn, :passive) do
+    else
+      {:error, %Mint.TransportError{reason: :nxdomain}} ->
+        false
+    end
```

Comment:
We don't want to crash in case anything else goes wrong

```suggestion
         {:ok, _} <- Mint.HTTP.set_mode(conn, :passive),
         do: true,
         else: (_ -> false)
```